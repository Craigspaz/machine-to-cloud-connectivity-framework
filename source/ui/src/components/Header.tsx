// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0


import { I18n } from '@aws-amplify/core';
import Button from 'react-bootstrap/Button';
import Navbar from 'react-bootstrap/Navbar';
import { signOut } from '../util/Utils';

/**
 * Renders the header of the UI.
 * @returns The header
 */
export default function Header(): JSX.Element {
  return (
    <header key="header">
      <Navbar bg="light" expand="sm">
        <Navbar.Brand href="/">{I18n.get('application')}</Navbar.Brand>
        <Navbar.Toggle />
        <Navbar.Collapse className="justify-content-end">
          <Button variant="link" onClick={signOut}>{I18n.get('sign.out')}</Button>
        </Navbar.Collapse>
      </Navbar>
    </header>
  );
}