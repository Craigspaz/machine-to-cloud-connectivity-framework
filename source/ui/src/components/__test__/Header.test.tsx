// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { I18n } from '@aws-amplify/core';
import { render } from '@testing-library/react';
import Header from '../Header';

I18n.setLanguage('en');

test('renders the Header component', async () => {
  const header = render(<Header />);
  expect(header.container).toMatchSnapshot();
});