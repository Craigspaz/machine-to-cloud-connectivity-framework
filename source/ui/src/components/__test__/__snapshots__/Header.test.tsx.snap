// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Header component 1`] = `
<div>
  <header>
    <nav
      class="navbar navbar-expand-sm navbar-light bg-light"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        application
      </a>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="justify-content-end navbar-collapse collapse"
      >
        <button
          class="btn btn-link"
          type="button"
        >
          sign.out
        </button>
      </div>
    </nav>
  </header>
</div>
`;
